addon/states.js

```diff
@@ -164,24 +164,23 @@ FragmentRootState = wireState(FragmentRootState, null, 'root');
 export default FragmentRootState;
 
 export function fragmentDidDirty(record, key, fragment) {
-  if (!get(record, 'isDeleted')) {
+  if (!record.currentState.isDeleted) {
     // Add the fragment as a placeholder in the owner record's
     // `_attributes` hash to indicate it is dirty
-    record._internalModel._attributes[key] = fragment;
-
+    record._internalModel._recordData.setDirtyAttribute(key, fragment);
     record.send('becomeDirty');
   }
 }
 
 export function fragmentDidReset(record, key) {
   // Make sure there's no entry in the owner record's
   // `_attributes` hash to indicate the fragment is dirty
-  delete record._internalModel._attributes[key];
+  delete record._internalModel._recordData._attributes[key];
 
   // Don't reset if the record is new, otherwise it will enter the 'deleted' state
   // NOTE: This case almost never happens with attributes because their initial value
   // is always undefined, which is *usually* not what attributes get 'reset' to
-  if (!get(record, 'isNew')) {
+  if (!record.currentState.isNew) {
```

Comment:
Any potential issues with accessing `currentState` directly? Would using `internalModel._currentState` be a better idea?

Reply:
It seems like it should be ok